### PR TITLE
first draft

### PR DESCRIPTION
**Pull request by nerisa**
@leapfrog-dipendra  @adhpawal  @bhuwang @anishmanandhar @Satish-Gupta @utsabneupane @Msanjib

### CODE REVIEW

File: java_training/src/dec29/Calculator.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+			case "5":	modulus();
+						break;
+			default:	System.out.println("Invalid choice");
+						break;
+			
+		}
+		choice.close();
+		
+	}
+	public static void add(){
```
Comment (adhpawal) [inline java_training/src/dec29/Calculator.java]: Define Method level JavaDoc as well. 

See this Class and it's JavaDoc http://www.docjar.net/html/api/java/util/Collections.java.html

So the common format for the JavaDoc should at minmum look like ,

```
/**
Method Description.

@author
@param param1 : {@link DataType}
@param param2 : {@link DataType}
@return DataType

**/

```

File: java_training/src/dec29/Calculator.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+package dec29;
+
+import java.util.Scanner;
+
+/**
+ * @author nerisa
+ *
+ */
+public class Calculator {
+	static Scanner userInput1=new Scanner(System.in);
```
Comment (adhpawal) [inline java_training/src/dec29/Calculator.java]: So, Why exaclty do you need two `Scanner` here?

File: java_training/src/dec29/Calculator.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	static Scanner userInput2=new Scanner(System.in);
+	static double firstNumber,secondNumber;
```
Comment (adhpawal) [inline java_training/src/dec29/Calculator.java]: And why `static` declaration for these ?

Comment (nerisa) [inline java_training/src/dec29/Calculator.java]: @adhpawal I need to call those functions from the main which is static so I declared them static

File: java_training/src/dec29/Calculator.java (hunk 2 above)
Comment (nerisa) [inline java_training/src/dec29/Calculator.java]: @adhpawal I thought same Scanner variable cannot be used for two simultaneous inputs. But now I realized its not the case so its not necessary

File: java_training/src/dec29/Calculator.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+		String userChoice;
+		System.out.println(	"1. Add\n"+
+							"2. Subtract\n"+
+							"3. Multiply\n"+
+							"4. Divide\n"+
+							"5. Remainder");
+		System.out.println("Enter the required operation:");
+		userChoice=choice.nextLine();
+		switch(userChoice){
+			case "1":  	add();
```
Comment (anishmanandhar) [inline java_training/src/dec29/Calculator.java]: Good work :+1:  , Defining separate methods for each oepration :)

File: java_training/src/dec29/Calculator.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+						break;
+			case "4":	divide();
+						break;
+			case "5":	modulus();
+						break;
+			default:	System.out.println("Invalid choice");
+						break;
+			
+		}
+		choice.close();
```
Comment (anishmanandhar) [inline java_training/src/dec29/Calculator.java]: handle these operations with try catch block and put choice.close() in the finally block

File: java_training/src/dec29/Calculator.java [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+		
+	}
+	public static void add(){
+		double result=0.0;
+		System.out.println("Enter the two nos to be added");
+		try{
+			firstNumber=Double.parseDouble(userInput1.nextLine());
+			secondNumber=Double.parseDouble(userInput2.nextLine());
+		}catch (NumberFormatException e){
+			System.out.println("Invalid numbers");
```
Comment (anishmanandhar) [inline java_training/src/dec29/Calculator.java]: Print error message too and use System.out.err("Expcetion while formatting number => "+ e.getMessage());

File: java_training/src/dec29/MultipleFinder.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+ * @author nerisa
+ *
+ */
+public class MultipleFinder {
+
+	/**
+	 * @param args
+	 */
+	public static void main(String[] args) {
+		int startingPoint=100,sum=0,count=0;
```
Comment (anishmanandhar) [inline java_training/src/dec29/MultipleFinder.java]: This is not good practice to declare all variable at a single line although it works.
Please declare variables in separate line which seems more readable

File: java_training/src/dec29/Calculator.java (hunk 3 above)
Comment (nerisa) [inline java_training/src/dec29/Calculator.java]: @adhpawal Is there another way without declaring them static?

File: java_training/src/dec29/Calculator.java [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+		}
+		result=firstNumber+secondNumber;
+		System.out.printf("Result= %f", result);
```
Comment (adhpawal) [inline java_training/src/dec29/Calculator.java]: Not necessayr but you mcould simply use System.out.println("Result= ", result); without the necessity of format.

File: java_training/src/dec29/PalindromeChecker.java [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+			reverseNumber = reverseNumber + initialNumber.charAt(i);
+		}
+		// System.out.println(reverseNumber);
+		if (initialNumber.equals(reverseNumber)) {
+			System.out.println("Palindrome");
+		} else {
+			System.out.println("Not a Palindrome");
+		}
+	}
+
```
Comment (anishmanandhar) [inline java_training/src/dec29/PalindromeChecker.java:68]: What will be the result if i enter -12321
Is this only for positive numbers?

File: java_training/src/dec29/Calculator.java [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+		double result=0.0;
+		try{
+			System.out.println("Enter the dividend");
+			firstNumber=Double.parseDouble(userInput1.nextLine());
+			System.out.println("Enter the divisor");
+			secondNumber=Double.parseDouble(userInput2.nextLine());
+		}catch (NumberFormatException e){
+			System.out.println("Invalid numbers");
+		}
+		result=firstNumber/secondNumber;
```
Comment (adhpawal) [inline java_training/src/dec29/Calculator.java]: What if `secondNumber` is 0 or both values are 0.

File: java_training/src/dec29/MultipleFinder.java [hunk 11]
```diff
@@ -0,0 +1,29 @@
+/**
+ * 
+ */
+package dec29;
+
+/**
+ * @author nerisa
+ *
+ */
+public class MultipleFinder {
```
Comment (anishmanandhar) [inline java_training/src/dec29/MultipleFinder.java:10]: No Comment??
What is the purpose of this class ? :)

File: java_training/src/dec29/Calculator.java [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+		double result=0.0;
+		try{
+			System.out.println("Enter the dividend");
+			firstNumber=Double.parseDouble(userInput1.nextLine());
+			System.out.println("Enter the divisor");
+			secondNumber=Double.parseDouble(userInput2.nextLine());
+		}catch (NumberFormatException e){
+			System.out.println("Invalid numbers");
+		}
+		result=firstNumber%secondNumber;
```
Comment (adhpawal) [inline java_training/src/dec29/Calculator.java]: What if secondNumber is 0 or both values are 0.

File: java_training/src/dec29/PalindromeChecker.java [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+ */
+public class PalindromeChecker {
+	static int number = 0;
+
+	public static boolean validChecker() {
+		boolean inputValid = false;
+		if (Integer.toString(number).length() == 5) {
+			inputValid = true;
+		}
+		return inputValid;
```
Comment (adhpawal) [inline java_training/src/dec29/PalindromeChecker.java]: or to sum up you could simply do , 

```
return Integer.toString(number).length();

```

But, In second thought the first one is OK on readability

File: java_training/src/dec29/PalindromeChecker.java [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+					}
+				} while (notInteger);
+				is5digit = validChecker();
+			} while (!is5digit);
+		} finally {
+			userInput.close();
+		}
+		initialNumber = Integer.toString(number);
+		// System.out.println(initialNumber);
+		for (int i = initialNumber.length() - 1; i >= 0; i--) {
```
Comment (adhpawal) [inline java_training/src/dec29/PalindromeChecker.java:59]: So, I guess you have already used StringBuilder previously. So, you could probably use something like , 

 `StringBuilder(initialNumber).reverse()`

i hven't checked i. But it is defintely there.

Soemtime it is better to explore all these functionality.

File: java_training/src/dec29/TemperatureDisplayer.java [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
+
+	/**
+	 * @param args
+	 */
+	public static void main(String[] args) {
+		double celsius=0.0,fahrenheit=0.0;
+		do{
+			fahrenheit=celsius *(9.0/5.0)+32;
+			// \u2103=degree celsius symbol \u2109=degree fahrenheit symbol
+			System.out.printf("%.2f \u2103= %.2f \u2109\n",celsius,fahrenheit);
```
Comment (adhpawal) [inline java_training/src/dec29/TemperatureDisplayer.java]: So i guess \u2103 is degreee symbol then :D

File: java_training/src/dec29/Calculator.java (hunk 8 above)
Comment (nerisa) [inline java_training/src/dec29/Calculator.java]: I got the error message 'The method println(double) in the type PrintStream is not applicable for the arguments (String, double)' while trying to do that so I changed it to printf

Comment (nerisa) [inline java_training/src/dec29/Calculator.java]: I left out the '+' in before the result. That's why I got that error:D

File: java_training/src/dec29/Calculator.java [hunk 16]
```diff
@@ -0,0 +1,135 @@
...
+	 * @author nerisa
+	 * @param firstNumber operand of the operations
+	 * @param secondNumber operand of the operations
+	 */
+	static Scanner userInput=new Scanner(System.in);
+	static double firstNumber,secondNumber;
+	public static void main(String[] args) {
+		Scanner choice=new Scanner(System.in);
+		String userChoice;
+		System.out.println(	"1. Add\n"+
```
Comment (amannepid) [inline java_training/src/dec29/Calculator.java]: proper code formatting needed. look for code format in eclipse or shortcut key

File: java_training/src/dec29/TemperatureDisplayer.java [hunk 17]
```diff
@@ -0,0 +1,29 @@
+/**Write a program that displays the temperatures from 0 degrees Celsius to 100 degrees Celsius 
+ * and its Fahrenheit equivalent. 
+ * Note that the conversion from Celsius to Fahrenheit uses the following formula: 
+ * F = C * 9/5 + 32;
+ * 
+ */
```
Comment (amannepid) [inline java_training/src/dec29/TemperatureDisplayer.java]: class description should be above the class declaration

**Pull request merged**